### PR TITLE
Refactor statsbeat

### PR DESCRIPTION
**Pull request by vishweshbankwar**
Refactoring to move initialization from static constructor to a static method.

### CODE REVIEW

**Review by rajkumar-rangaraj — APPROVED** [review]

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/Statsbeat/Statsbeat.cs [hunk 1]
```diff
@@ -106,11 +121,24 @@ private static VmMetadataResponse GetVmMetadataResponse()
             }
             catch (Exception ex)
             {
-                AzureMonitorExporterEventSource.Log.WriteWarning("Failed to get VM metadata details", ex);
+                AzureMonitorExporterEventSource.Log.WriteInformational("Failed to get VM metadata details", ex.ToInvariantString());
```
Comment (TimothyMothra) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/Statsbeat/Statsbeat.cs:124]: I wouldn't do the `ToInvariantString` here.
Better to add a new `NonEvent` method to the EventSource class to handle this.
The main benefit is to potentially skip the work if noone is listening to the eventsource.

Comment (vishweshbankwar) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/Statsbeat/Statsbeat.cs:124]: Will fix this.

Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/Statsbeat/Statsbeat.cs:124]: On second thought, maybe we leave this as Warning. :)

1. The original idea was that Informational logs wouldn't contain exceptions.
2. The AzureMetaData service won't be available to non-Azure environments. 
    If we get an Exception in Azure, the warning would be valid.
    Outside of Azure, I think the warning is noise which is why I originally suggested using Informational.
    
@rajkumar-rangaraj what do you think?

Comment (cijothomas) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/Statsbeat/Statsbeat.cs:124]: > The main benefit is to potentially skip the work if noone is listening to the eventsource.

^ This code is executed only once, right?

**Pull request merged**